cli/azd/cmd/init.go

```diff
@@ -324,5 +332,18 @@ func (i *initAction) Run(ctx context.Context) error {
 		return fmt.Errorf("saving default environment: %w", err)
 	}
 
+	// If the configuration is empty, set default subscription & location
+	// This will be the case for first run experience
+	if i.config.IsEmpty() {
+		_, err = i.accountManager.SetDefaultSubscription(ctx, env.GetSubscriptionId())
+		if err != nil {
+			return fmt.Errorf("failed setting default subscription. %w", err)
+		}
+		_, err = i.accountManager.SetDefaultLocation(ctx, env.GetSubscriptionId(), env.GetLocation())
+		if err != nil {
+			return fmt.Errorf("failed setting default location. %w", err)
+		}
+	}
+
```

Comment:
Should this belong closer to where we want to set the location? Functionally, I'd expect the defaulting logic to be close to where the user selected the prompt. 

This will likely get lost in a refactor once we move environment initialization later into `azd provision/deploy` as needed.

Reply:
See similar comment below.  We only want to save these defaults on our first run when a user does not have any configuration.

Reply:
I don't really think of `azd init` as the forced starting point for the user. A user can run `azd provision` and `azd deploy`, and never touch `azd init`, since they're using their own repository or cloning via github or forking. 

To me, the "first-run" experience means: when the user selects a subscription and location from the prompt and does not yet have the values defined in the config file. Thus, I would save the default values if not present when a selection is made in the location prompt. I think this also avoids this weird side-effect of `azd init` persisting the defaulted values when it doesn't really use the subscription/location values or the config.

Reply:
Replied in new thread. Closing this one.